src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,335 @@
+#   logs_dir: path to local logs
+#   threshold: value to use as threshold to classify big files
+# Returns:
+#   stdout output listing the files prepended with a 'Y' or 'N' tag
+#######################################
+list_tag_files() {
+    # shellcheck disable=SC2016,SC1004,SC2086
+    awk_cmd='{ \
+    if (th <= $1){ \
+        print "  Y:",$1,$2} \
+    else { \
+        print "  N:",$1,$2}\
+    }'
+
+    # Convert to bytes and run
+    bts=$(echo "${2}" | numfmt --from=iec)
+    # shellcheck disable=SC2086
+    du -ab -d 1 ${1} | awk -v th="${bts}" "${awk_cmd}"
+}
+
+#######################################
+# Check if this host is a VM.
+# Arguments:
+#   $1: whether or not to skip the check
+# Returns:
+#   0: VM detected
+#   1: No VM detected or skipped
+#######################################
+check_hw() {
+    if dmesg | grep -q "Hypervisor detected"; then
+        # This host is a VM
+        if "${1:-false}"; then
+            echo "  Running compression on a VM host"
+            return 1
+        fi
+        return 0
+    fi
+    return 1
+}
+
+#######################################
+# Compress any files exceeding 1M.
+# Arguments:
+#   $1: local logs to compress
+# Returns:
+#   status of the lbzip2 command
+#######################################
+compress_files() {
+    # shellcheck disable=SC2086
+    find ${1} -maxdepth 0 -type f -size +1M -print0 | xargs -r0 lbzip2 -v
+    return $?
+}
+
+#######################################
+# Archive files to the remote host.
+# Arguments:
+#   $1: local logs to copy
+#   $2: remote location to copy files
+# Returns:
+#   0: All commands were successful
+#   1: Failure detected with at least one command
+#######################################
+scp_files() {
+    rc=0
+    copied=()
+    # shellcheck disable=SC2045,SC2086
+    for file in $(ls -d ${1})
+    do
+        file_name=${file##*/}
+        archive_name="${file_name%%.*}.$(hostname -s).${file_name#*.}"
+        if scp -r "${file}" "${2}"/"${archive_name}"; then
+            copied+=("${file}")
+            if ! rm -fr "${file}"; then
+                echo "  Error removing ${file}"
+                echo "    $(ls -al ${file})"
+                rc=1
+            fi
+        else
+            echo "  Failed to archive ${file} to ${2}"
+            rc=1
+        fi
+    done
+    echo "  The following files were archived:"
+    for file in "${copied[@]:-no files}"
+    do
+        echo "    ${file}"
+    done
+    return ${rc}
+}
+
+#######################################
+# Run cart_logtest.py against the logs.
+# Arguments:
+#   $1: local logs
+# Returns:
+#   0: All cart_logtest were successful
+#   1: Failure detected with cart_logtest
+#######################################
+run_cartlogtest() {
+    rc=0
+    if [ -f "${CART_LOGTEST_PATH}" ]; then
+        # shellcheck disable=SC2045,SC2086
+        for file in $(ls -d ${1})
+        do
+            if [ -f ${file} ] && [[ ! ${file} =~ "cart_logtest" ]]; then
```

Comment:
(lint) Don't quote rhs of =~, it'll match literally rather than as a regex. [SC2076]